### PR TITLE
Adding in docker image for chatops installation

### PR DESCRIPTION
**Pull request by Xorso**
This is to add another image for the docker setup to enable the chatops if it is desired.

Running:
` make build-chatops`

In order for the service to come online the configuration needs to be completed.

### CODE REVIEW

**Comment by shusugmt** [comment]
Thanks for the PR.

I see that this `Dockefile` is almost identical to the original `Dockerfile` except the following lines:

```
$ diff stackstorm/Dockerfile stackstorm-chatops/Dockerfile
7a8
>
62a64,67
> # Install st2-chatops with Node.js v4 requirement
> RUN curl -sL https://deb.nodesource.com/setup_4.x | sudo -E bash -
> RUN sudo apt-get install -y st2chatops
>
```

Can't we just start `FROM stackstorm/stackstorm:latest` and put those only three lines to add `st2chatops` on top of the _base_ image so that we can avoid keeping multiple copies of the exact same code fragment?

@warrenvw Any thoughts?

**Comment by warrenvw** [comment]
I concur. 👍

**Comment by Xorso** [comment]
I sure can. In the chatrooms it made it sound like there wanted to be a different image for having chatops enabled. I can issue a PR for the same image with those changes.

**Comment by warrenvw** [comment]
Another possibility is installing chatops in the `stackstorm` image, and only enabling it to start on boot via an `/entrypoint.d` script. Perhaps move your changes to `images/stackstorm/Dockerfile`, along with the following to disable st2chatops:

```
echo manual | sudo tee /etc/init/st2chatops.override
```

Simplify remove the override file if you want to start st2chatops.

Having said this, I'm okay with previous suggestion if it makes more sense.

**Comment by shusugmt** [comment]
> installing chatops in the stackstorm image, and only enabling it to start on boot via an /entrypoint.d script

This seems to be better. @Xorso can you fix the code?

**Comment by Xorso** [comment]
Give me a day or say and I will get the PR updated to use entrypoint.d

**Comment by Xorso** [comment]
Let me know if those changes look good.

File: images/stackstorm/Dockerfile [hunk 1]
```diff
@@ -69,6 +69,12 @@ RUN sed -i 's/start on filesystem and net-device-up IFACE!=lo/start on runlevel
     && sed -i '/start mistral-api/i\  /opt/stackstorm/mistral/bin/mistral-db-manage --config-file /etc/mistral/mistral.conf upgrade head\n\
   /opt/stackstorm/mistral/bin/mistral-db-manage --config-file /etc/mistral/mistral.conf populate\n' /etc/init/mistral.conf
 
+# Install chatops and disable unless entrypoint.d file is present
+# Install st2-chatops with Node.js v4 requirement
+RUN curl -sL https://deb.nodesource.com/setup_4.x | sudo -E bash -
+RUN sudo apt-get install -y st2chatops
+RUN echo manual | sudo tee /etc/init/st2chatops.override
```
Comment (shusugmt) [inline images/stackstorm/Dockerfile]: Pack these three commands into a single `RUN` statement

**Comment by Xorso** [comment]
I am not sure if I squashed those right or not. Please verify and let me know if I need to change something.

**Comment by shusugmt** [comment]
Closed since st2chatops is already in master.

**Pull request merged**